sirepo/template/elegant.py

```diff
@@ -166,6 +166,58 @@ def _lattice_filename(self, value):
         r.output_files = [f[k] for k in f.keys_in_order]
         return r
 
+    def _convert(self, data):
+
+        s = _SCHEMA.model
+        c = _code_var(data.models.rpnVariables)
+        for x in  data.models.rpnVariables:
+            x.value = c.eval_var_with_assert(x.value)
+
+        def _model(model, name):
+            schema = s[name]
+            t = None
+
+            def _rpn(rpn_type, py_type):
+                nonlocal t
+                if (
+                    t != rpn_type or
+                    not code_variable.CodeVar.is_var_value(v)
+                ):
+                    return False
+                model[k] = c.eval_var_with_assert(v)
+                return True
+
```

Comment:
The embedded functions make the logic hard to follow and requires hunting for the source values like "v". I much prefer passing values to the function. Two levels of embedded functions makes this worse.

Reply:
I removed the `_rpn()` because it was only used in the `RPNValue` case. This eliminated the second nesting.  

I think `nonlocal` wasn't working right so after making that change, values of `RPNValue` were getting converted to floats, and this lead to some interesting issues like defaults don't match. Also, clearly integer values like `bins` are now floats.